pkg/issuer/acme/dns/dns.go

```diff
@@ -91,39 +91,29 @@ func (s *Solver) Present(ctx context.Context, issuer v1alpha1.GenericIssuer, ch
 }
 
 // Check verifies that the DNS records for the ACME challenge have propagated.
-func (s *Solver) Check(ctx context.Context, issuer v1alpha1.GenericIssuer, ch *v1alpha1.Challenge) (bool, error) {
-	providerName := ch.Spec.Config.DNS01.Provider
-	if providerName == "" {
-		return false, fmt.Errorf("dns01 challenge provider name must be set")
-	}
+func (s *Solver) Check(ctx context.Context, issuer v1alpha1.GenericIssuer, ch *v1alpha1.Challenge) error {
 
-	providerConfig, err := issuer.GetSpec().ACME.DNS01.Provider(providerName)
+	fqdn, value, ttl, err := util.DNS01Record(ch.Spec.DNSName, ch.Spec.Key, s.DNS01Nameservers, false)
```

Comment:
As discussed on Slack, the change from `followCNAME(providerConfig.CNAMEStrategy)` to `false` makes sense as we should be relying on the recursive DNS server to resolve CNAMEs for us anyway, so we needn't ever resolve CNAMEs ourselves as part of the DNS01 self check.